### PR TITLE
Error if pocketfft is not found

### LINKED ISSUE #67842: Pytorch 1.10.0 missing fft support on android? RuntimeError: fft: ATen not compiled with FFT support
## 🐛 Bug
Hi everyone,

I am currently trying to port [silero-vad](https://github.com/snakers4/silero-vad) to android (mentioning @snakers4 FYI). The app is still in an early stage and I have zero experience with android. It is highly likely that it isn't a bug and I am just doing something wrong, but it looks like the android PyTorch builds do not have fft support. 

According to the [release notes](https://github.com/pytorch/pytorch/releases/tag/v1.10.0) (Misc section) of 1.10.0, PyTorch for arm should be compiled with fft support but I receive the following error messages during the forward pass:

org.pytorch:pytorch_android_lite:1.10.0:
```
E/AndroidRuntime: FATAL EXCEPTION: Thread-2
    Process: com.example.testVAD, PID: 6113
    com.facebook.jni.CppException: fft: ATen not compiled with FFT support
      
      Debug info for handle(s): -1, was not found.
      
    Exception raised from _fft_r2c_mkl at ../aten/src/ATen/native/mkl/SpectralOps.cpp:569 (most recent call first):
    (no backtrace available)
        at org.pytorch.LiteNativePeer.forward(Native Method)
        at org.pytorch.Module.forward(Module.java:52)
        at com.example.testVAD.MainActivity.recognize(MainActivity.java:210)
        at com.example.testVAD.MainActivity.run(MainActivity.java:188)
        at java.lang.Thread.run(Thread.java:919)
```
org.pytorch:pytorch_android:1.10.0
```
E/AndroidRuntime: FATAL EXCEPTION: Thread-2
    Process: com.example.testVAD, PID: 10231
    java.lang.RuntimeException: The following operation failed in the TorchScript interpreter.
    Traceback of TorchScript, serialized code (most recent call last):
      File "code/__torch__/models/number_vad_model.py", line 17, in forward
        x: Tensor) -> Tensor:
        feature_extractor = self.feature_extractor
        x0 = (feature_extractor).forward(x, )
              ~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        adaptive_normalization = self.adaptive_normalization
        x1 = (adaptive_normalization).forward(x0, )
      File "code/__torch__/models/number_vad_model.py", line 42, in forward
        win_length = self.win_length
        hop_length = self.hop_length
        stft_sample = __torch__.torch.functional.stft(wav, n_fft, hop_length, win_length, None, True, "reflect", False, None, None, )
                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ <--- HERE
        mag, phase, = _1(stft_sample, 1., )
        mode = self.mode
      File "code/__torch__/torch/functional.py", line 21, in stft
      else:
        input0 = input
      _2 = torch.stft(input0, n_fft, hop_length, win_length, window, normalized, onesided, return_complex)
           ~~~~~~~~~~ <--- HERE
      return _2
    
    Traceback of TorchScript, original code (most recent call last):
      File "/opt/conda/lib/python3.8/site-packages/torch/functional.py", line 515, in stft
            input = F.pad(input.view(extended_shape), (pad, pad), pad_mode)
            input = input.view(input.shape[-signal_dim:])
        return _VF.stft(input, n_fft, hop_length, win_length, window,  # type: ignore
               ~~~~~~~~ <--- HERE
                        normalized, onesided, return_complex)
    RuntimeError: fft: ATen not compiled with FFT support
    
        at org.pytorch.NativePeer.forward(Native Method)
        at org.pytorch.Module.forward(Module.java:52)
        at com.example.testVAD.MainActivity.recognize(MainActivity.java:210)
        at com.example.testVAD.MainActivity.run(MainActivity.java:188)
        at java.lang.Thread.run(Thread.java:919)
```

## To Reproduce

On an x86 machine:
```
import torch
model, utils = torch.hub.load(repo_or_dir='snakers4/silero-vad', model='silero_vad', force_reload=True)
model._save_for_lite_interpreter("AndroidStudioProjects/testVAD/app/src/main/assets/silero_vad.ptl")
```
 *I started with the [s2t demo](https://github.com/pytorch/android-demo-app/tree/master/SpeechRecognition) and can load the model but the forward pass leads to the error message above.
Android app:
```
//loading the model
    private void load_encoder() throws FileNotFoundException{
        String silero_vad_path = null;
        File file = new File(getApplicationContext().getFilesDir(), "silero_vad.ptl");

        try (InputStream is = getApplicationContext().getAssets().open("silero_vad.ptl")) {
            OutputStream os = new FileOutputStream(file);
            byte[] buffer = new byte[4 * 1024];
            int read;
            while ((read = is.read(buffer)) != -1) {
                os.write(buffer, 0, read);
            }
            os.flush();

            silero_vad_path = file.getAbsolutePath();
        } catch (IOException e) {
            throw new FileNotFoundException("silero_vad pytorch model not found");
        }

        //vad_encoder = LiteModuleLoader.load(silero_vad_path);
        vad_encoder = Module.load(silero_vad_path);
    }


    private String recognize(float[] floatInputBuffer){
        double audio_input[] = new double[RECORDING_LENGTH];
        for (int n = 0; n < RECORDING_LENGTH; n++)
            audio_input[n] = floatInputBuffer[n];

        FloatBuffer inTensorBuffer = Tensor.allocateFloatBuffer(RECORDING_LENGTH);
        for (double val : audio_input)
            inTensorBuffer.put((float)val);

        Tensor inTensor = Tensor.fromBlob(inTensorBuffer, new long[]{1, RECORDING_LENGTH});
        final String result = vad_encoder.forward(IValue.from(inTensor)).toStr();

        return result;
    }
```

## Expected behavior

Pytorch Android packages should be built with fft support or in case the error is different, provide a better error message.

## Environment

 - PyTorch Version (e.g., 1.0): 1.10.0+cpu
 - OS (e.g., Linux): Ubuntu 20.04
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.8
 - CUDA/cuDNN version:-
 - GPU models and configuration:-
 - Any other relevant information:

@peterbell10 could you please have a look?

cc @mruberry @peterbell10

### PR DESCRIPTION
**Pull request by peterbell10**
Fixes #67842

cc @mruberry @peterbell10

### CODE REVIEW

File: cmake/Dependencies.cmake [hunk 1]
```diff
@@ -293,15 +293,15 @@ endif()
...
 if(NOT AT_MKL_ENABLED)
-  find_path(POCKETFFT_INCLUDE_DIR NAMES pocketfft_hdronly.h PATHS
-            /usr/local/include
-            ENV POCKETFFT_HOME
-            "${PROJECT_SOURCE_DIR}/third_party/pocketfft"
-           )
-  if(POCKETFFT_INCLUDE_DIR)
-    set(AT_POCKETFFT_ENABLED 1)
-    message(STATUS "Using pocketfft in directory: ${POCKETFFT_INCLUDE_DIR}")
+  set(POCKETFFT_INCLUDE_DIR "${Torch_SOURCE_DIR}/third_party/pocketfft/")
```
Comment (mruberry) [inline cmake/Dependencies.cmake:296]: Help me out with this logic, @peterbell10 

So with this change if AT_MKL_ENABLED is false, then pocketfft must be available in `Torch_SOURCE_DIR/third_party/pocketfft`?

Comment (peterbell10) [reply cmake/Dependencies.cmake:296]: It's already a submodule, so as long as you do `git submodule init` then this works.

Comment (peterbell10) [reply cmake/Dependencies.cmake:296]: Incase you're not familiar, `${Torch_SOURCE_DIR}` is the pytorch root directory.

**Review by malfet — CHANGES_REQUESTED** [review]
I'm fine with the change on its own, but I don't believe it fixes the issue, as PyTorch mobile builds already checkout all submodules, but are still build without PocketFFT support. (And correct me if I'm wrong, but this change is no-op if submodule is present, isn't it?)
But I could be totally wrong, so could you please describe in PR description how this fix was tested?

**Comment by peterbell10** [comment]
The first commit of this PR failed in CI e.g. [here](https://app.circleci.com/pipelines/github/pytorch/pytorch/405581/workflows/281b7a8a-13bc-4781-b0d3-e959c0f1d14d/jobs/16673715) so clearly the  `find_path` was failing for some reason.

**Comment by peterbell10** [comment]
@malfet ping

**Comment by malfet** [comment]
@peterbell10 I'm a bit confused by this explanation. Are you saying we should not be using `find_path` at all in our codepath? Or that it for some unknown reason does not work just for android builds?

**Comment by peterbell10** [comment]
>Are you saying we should not be using `find_path` at all in our codepath? Or that it for some unknown reason does not work just for android builds?

Yes, `find_path` was not working for the android builds. I'm not sure exactly why not, but the issue was visible in CI so the checks I've added are covering the cases that were failing previously with `find_path`.

**Comment by peterbell10** [comment]
FWIW I can't reproduce it locally, but #66543 includes several users saying that pocketfft isn't being picked up by CMake.

**Comment by peterbell10** [comment]
@malfet ping

**Comment by peterbell10** [comment]
@malfet ping

**Comment by peterbell10** [comment]
@malfet ping

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**